### PR TITLE
Rewire deliverable APIs to the backend 

### PR DESCRIPTION
**Pull request by nickgraz**
- Use latest backend types
- Update these APIs to use the backend instead of mock data:
  - List deliverables
  - Get deliverable
  - Update deliverable

TODO:
[x] Rename 'reason' to 'feedback' in a few other places
[x] Hook up the `update` call
[x] Figure out where the `templateUrl` is supposed to come from 
[x] Search and sort on deliverables on the frontend (in #2242)

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* **#2235** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2235?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2235?utm_source=stack-comment).

File: src/services/DeliverablesService.ts [hunk 1]
```diff
@@ -1,315 +1,69 @@
...
-    return result;
-  }
-
-  return result.map((element) => ({
-    ...transformAdminDeliverableElement(element),
-    description: `${element.description}`,
-  }));
-};
+  searchSortOrder?: SearchSortOrder
+  // TODO sort, search, etc...
```
Comment (karthikbtf) [inline src/services/DeliverablesService.ts:52]: If we will be doing client site sorting, there might be some code [here](https://github.com/terraware/terraware-web/blob/main/src/scenes/PlantingSitesRouter/view/PlantingSitesList.tsx#L114) to extract out as reusable. Handles string vs numeric from the sort order object.
We may need to get number properties localized and in the raw format.

**Review by karthikbtf — COMMENTED** [review]
Thanks for taking this on. Big step towards integration.

File: src/services/DeliverablesService.ts (hunk 1 above)
Comment (nickgraz) [reply src/services/DeliverablesService.ts:65]: Cool, working on that now, will put in another PR

**Review by karthikbtf — APPROVED** [review]
Woot 🎉 🙌

**Comment by nickgraz** [comment]
### Merge activity

* **Feb 29, 6:45 PM EST**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2235).

**Pull request merged**